src/Core/tests/DeviceTests.Shared/HandlerTests/HandlerTestBase.cs

```diff
@@ -78,8 +78,8 @@ protected void InitializeViewHandler(IElement element, IElementHandler handler,
 		{
 			mauiContext ??= MauiContext;
 			handler.SetMauiContext(mauiContext);
-			handler.SetVirtualView(element);
 			element.Handler = handler;
+			handler.SetVirtualView(element);
```

Comment:
@PureWeen I spotted this. In the real code, we set `Handler` first. It appears that `SetVirtualView` also sets things, but those need to run later? Not quite sure how this is as both set the Handler. But it resulted in the OnHandlerChangedCore running before the container views were created.